threedi_schema/migrations/versions/0300_upgrade_db_settings.py

```diff
@@ -0,0 +1,296 @@
+
+
+def remove_columns_from_table(table_name: str, columns: List[str]):
+    # no checks for existence are done, this will fail if any table or column doesn't exist
+    with op.batch_alter_table(table_name) as batch_op:
+        for column in columns:
+            batch_op.drop_column(column)
+
+
+def set_use_from_settings_id():
+    for settings_col, settings_id, settings_table in GLOBAL_SETTINGS_ID_TO_BOOL:
+        if settings_col != '':
+            # set boolean 'use_*' in model_settings if a relationship exists
+            op.execute(f"UPDATE model_settings SET {settings_col} = TRUE WHERE {settings_id} IS NOT NULL;")
+        # remove all settings rows, exact for the one matching
+        # delete_all_but_matching_id(settings_table, settings_id)
+        op.execute(
+            f"DELETE FROM {settings_table} WHERE NOT EXISTS "
+            f"(SELECT 1 FROM model_settings WHERE model_settings.{settings_id} IS NOT NULL);")
+        # command above doesn't catch id = None, so an extra command is needed for those cases
+        op.execute(
+            f"DELETE FROM {settings_table} WHERE id != "
+            f"(SELECT {settings_id} FROM model_settings);")
+
+    # set use_groundwater_flow
+    sql = """
+        UPDATE model_settings
+        SET use_groundwater_flow = CASE
+            WHEN (SELECT groundwater_hydraulic_conductivity FROM groundwater LIMIT 1) IS NOT NULL OR (SELECT groundwater_hydraulic_conductivity_file FROM groundwater LIMIT 1) IS NOT NULL THEN 1
+            ELSE 0
+        END;
+        """
+    op.execute(sql)
+
+
+def drop_columns(table, columns):
+    with op.batch_alter_table(table) as batch_op:
+        for column in columns:
+            batch_op.drop_column(column)
+
+
+def set_use_from_exists(table, settings_col):
+    op.execute(f"UPDATE model_settings SET {settings_col} = TRUE WHERE EXISTS (SELECT 1 FROM {table});")
+
+
+def delete_all_but_matching_id(table, settings_id):
+    op.execute(f"DELETE FROM {table} WHERE id NOT IN (SELECT {settings_id} FROM model_settings);")
+
+
+def delete_all_but_first_row(table):
+    op.execute(f"DELETE FROM {table} WHERE id NOT IN ("
+               f"SELECT id FROM "
+               f"(SELECT id FROM {table} ORDER BY id LIMIT 1) AS subquery);")
+
+
+def correct_dem_paths():
+    # remove path - this will not work with nested paths!
+    op.execute(f"UPDATE model_settings SET dem_file = SUBSTR(dem_file, INSTR(dem_file, '/') + 1)")
+
+
+def upgrade():
+    conn = op.get_bind()
+    # Only use first row of global settings
+    delete_all_but_first_row("v2_global_settings")
+    rename_tables(RENAME_TABLES)
+    # rename columns in renamed tables
+    for table_name, columns in RENAME_COLUMNS.items():
+        rename_columns(table_name, columns)
+    # make all columns in renamed tables, except id, nullable
+    for _, table_name in RENAME_TABLES:
+        make_all_columns_nullable(table_name)
+    # create new tables
+    create_new_tables(COPY_FROM_GLOBAL.keys())
```

Comment:
You can create tables and add the columns in one go (for SQLite, it makes no sense to first create tables because on each "add column" there is a new table created).

Example: https://github.com/nens/threedi-schema/blob/6b3b9dccba1b45fe84cebf81490b1acfb3de01a0/threedi_schema/migrations/versions/0216_vegetation_drag.py#L19